File: verify.js
const { chromium } = require('playwright');
const path = require('path');

(async () => {
    const browser = await chromium.launch();
    const page = await browser.newPage();

    // Log console errors to verify no `require is not defined`
    let errors = [];
    page.on('pageerror', err => {
        errors.push(err.message);
        console.error('Page error:', err.message);
    });

    await page.goto('http://localhost:8080');

    // Wait for the app to initialize
    await page.waitForSelector('#initial-loading-overlay.hidden', { timeout: 10000 }).catch(() => console.log('Overlay never hid'));

    // Check if the agent cards are rendered
    const cards = await page.$$('.flip-card:not(.skeleton-card)');
    console.log(`Found ${cards.length} loaded agent cards.`);

    if (errors.length > 0) {
        console.error(`Found ${errors.length} errors during initialization:`, errors);
        process.exit(1);
    }

    await browser.close();
    console.log('Frontend verified successfully.');
})();

File: verify.py
from playwright.sync_api import sync_playwright

def run():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        page = browser.new_page()

        page.goto("http://localhost:8080/index.html")
        page.wait_for_selector("#initial-loading-overlay", state="hidden")

        page.evaluate("""
            const modal = document.getElementById("fusionsModal");
            const listArea = document.getElementById("fusionsModalList");
            const downloadBtn = document.getElementById("downloadParentFusionsBtn");

            const listItems = `
                <li class="fusion-quick-list-item">
                    <button class="fusion-quick-btn" title="Bolt+">⚡</button>
                </li>
            `;
            listArea.innerHTML = `<ul class="fusion-quick-list-wrapper">${listItems}</ul>`;
            downloadBtn.style.display = "";
            modal.classList.add('visible');
        """)

        page.wait_for_timeout(500)
        page.screenshot(path="verification_fusion.png")

        browser.close()

if __name__ == "__main__":
    run()
